File: src/main/org/nlogo/api/PlotInterface.java
package org.nlogo.api;

public interface PlotInterface {
  void xMin_$eq(double xmin);

  void xMax_$eq(double xmax);

  void yMin_$eq(double ymin);

  void yMax_$eq(double ymax);

  void autoPlotOn_$eq(boolean autoPlot);

  void legendIsOpen_$eq(boolean open);

  void currentPen_$eq(String pen);

  scala.Option<PlotPenInterface> getPen(String pen);

  String name();

  void makeDirty();
}

File: src/main/org/nlogo/api/PlotInterface.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

trait PlotInterface {
  def xMin_=(xmin: Double): Unit
  def xMax_=(xmax: Double): Unit
  def yMin_=(ymin: Double): Unit
  def yMax_=(ymax: Double): Unit
  def autoPlotOn_=(autoPlot: Boolean): Unit
  def legendIsOpen_=(open: Boolean): Unit
  def currentPen_=(pen: String): Unit
  def getPen(pen: String): Option[PlotPenInterface]
  def name: String
  def makeDirty(): Unit
}
